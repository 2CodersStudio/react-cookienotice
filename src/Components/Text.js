// @flow

import * as React from 'react'
import styled from 'styled-components'
import clsx from 'clsx'

const TextStyled = styled.span`
  color: #000000;
  font-size: 16px;
  margin: 0 20px;

  &.dark {
    color: #ffffff;
  }

  @media (max-width: 768px) {
    font-size: 12px;
  }
`

type Props = {|
  label: string,
  darkTheme: boolean,
|}

export default class Text extends React.Component<Props> {
  render(): React.Node {
    const { label, darkTheme } = this.props

    return (
      <TextStyled
        className={clsx('react-cookienotice-text', { dark: darkTheme })}
      >
        {label}
      </TextStyled>
    )
  }
}
