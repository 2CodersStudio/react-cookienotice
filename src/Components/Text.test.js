// @flow

import * as React from 'react'
import { shallow } from 'enzyme'
import Text from './Text'

describe('Text', () => {
  describe('without darkTheme', () => {
    const comp = <Text label='label' darkTheme={false} />

    it('should render', () => {
      expect(comp).toMatchSnapshot()
    })

    it('should not have dark class', () => {
      const wrapper = shallow(comp)
      expect(wrapper.find('.dark')).toHaveLength(0)
    })
  })

  describe('with darkTheme', () => {
    const comp = <Text label='label' darkTheme />

    it('should render', () => {
      expect(comp).toMatchSnapshot()
    })

    it('should have dark class', () => {
      const wrapper = shallow(comp)
      expect(wrapper.find('.dark')).toHaveLength(1)
    })
  })
})
