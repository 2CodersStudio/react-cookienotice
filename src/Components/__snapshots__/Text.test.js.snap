// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text with darkTheme should render 1`] = `
<Text
  darkTheme={true}
  label="label"
/>
`;

exports[`Text without darkTheme should render 1`] = `
<Text
  darkTheme={false}
  label="label"
/>
`;
